pkg/otelcgen/otelcgen.go

```diff
@@ -238,6 +240,14 @@ func WithCA(caFile string) Option {
 	}
 }
 
+func WithSystemCAs(useSystemCAs bool) Option {
```

Comment:
Could you please add a unit test for the new field? I'm asking because moving it to another repo would be much easier in the future.

Reply:
Extended a unit test here [7c0cec8](https://github.com/Dynatrace/dynatrace-operator/pull/4630/commits/7c0cec88995c0f6d66887b740536c568b923e7c9)